packages/ext-angular-modern-kitchensink/src/app/view/landingpage/landingpage.component.ts

```diff
@@ -16,80 +29,76 @@ declare var Ext: any;
 })
 export class LandingpageComponent implements OnInit {
 
-  //code = window._code;
-
   ANGULAR_VERSION = VERSION.full;
 
-  nodeId;
-  selectedNavNode;
-  component;
   layout = "fit";
-  files;
-
-  nodeText;
-  nodeItems = [];
+  node: any;
+  node$: any = new Subject();
+  files: Array<String> = [];
+  breadcrumb: Array<any>;
 
   filterRegex;
   filterVal;
   showTreeFlag = true;
 
-  
+
   treeStore = Ext.create('Ext.data.TreeStore', {
     rootVisible: true,
     root: navTreeRoot
-  });  
+  });
+
+  constructor(location: Location, private router: Router, changeDetectorRef: ChangeDetectorRef, private resolver: ComponentFactoryResolver) {
+    this.node$.subscribe(({
+      next: (v) => {
+        this.node = v;
+        this.breadcrumb = generateBreadcrumb(v);
+        console.log(`BREADCRUMB: ${JSON.stringify(this.breadcrumb.map(b => b.text))}`);
+      },
+    }));
 
-  constructor(location: Location,private router: Router, changeDetectorRef: ChangeDetectorRef,
-    private resolver: ComponentFactoryResolver) { 
     router.events.subscribe((val) => {
-      if(val instanceof NavigationEnd) {
-        console.log(location.path(true));
-        console.log(val);
-        var path = location.path(true);
-        if(path){
-          this.nodeId = path.substring(path.indexOf("#")+1, path.length);
-          console.log("nodeId : " + this.nodeId);
-          this.selectedNavNode = this.treeStore.getNodeById(this.nodeId);
-          console.log(this.selectedNavNode);
-          if(this.selectedNavNode != null) {
-            this.nodeText = this.selectedNavNode.get('text');
-            console.log("this.nodeText : " + this.nodeText);
-            this.nodeItems.unshift(
-            );
-  
-            this.component = this.selectedNavNode.get('component');
-            console.log("Component: " + this.component);
-            if(this.selectedNavNode.get('layout') != null) {
-              this.layout = this.selectedNavNode.get('layout');
-              console.log("this.layout : " + this.layout);
-            }
-            //this.files = this.code[this.nodeText.replace(/\s/g, '')];
-            //console.log("this.files : " + this.files);
-          }
-          else {
-              console.log("selectedNavNode not found.")
-          }
+      if (val instanceof NavigationEnd) {
+        console.log(`location.path(true): ${location.path(true)}`);
+        console.log(`val: ${val}`);
+        console.log(`nodeId: ${val.url}`);
+        this.node$.next(this.treeStore.getNodeById(val.url));
```

Comment:
Why do we need this subscription model? Why not just get the node here and process?